packages/aws-rfdk/lib/lambdas/nodejs/sep-configuration/handler.ts

```diff
@@ -0,0 +1,288 @@
+              if ('ebs' in blockDeviceMapping) {
+                if ('deleteOnTermination' in blockDeviceMapping.ebs) {
+                  blockDeviceMapping.ebs.deleteOnTermination = this.convertToBoolean(blockDeviceMapping.ebs.deleteOnTermination);
+                }
+                if ('encrypted' in blockDeviceMapping.ebs) {
+                  blockDeviceMapping.ebs.encrypted = this.convertToBoolean(blockDeviceMapping.ebs.encrypted);
+                }
+                if ('iops' in blockDeviceMapping.ebs) {
+                  blockDeviceMapping.ebs.iops = Number.parseInt(blockDeviceMapping.ebs.iops, 10);
+                }
+                if ('volumeSize' in blockDeviceMapping.ebs) {
+                  blockDeviceMapping.ebs.volumeSize = Number.parseInt(blockDeviceMapping.ebs.volumeSize, 10);
+                }
+              }
+            });
+          }
+        });
+      }
+    }
+    return JSON.stringify(convertedSpotFleetRequestConfigs);
+  }
+
+  private spotFleetPluginConfigsToArray(pluginOptions: any): any {
+    let convertedConfigs = pluginOptions;
+    if ('DeleteInterruptedSlaves' in convertedConfigs) {
+      convertedConfigs.DeleteInterruptedSlaves = this.convertToBoolean(convertedConfigs.DeleteInterruptedSlaves);
+    }
+    if ('DeleteTerminatedSlaves' in convertedConfigs) {
+      convertedConfigs.DeleteTerminatedSlaves = this.convertToBoolean(convertedConfigs.DeleteTerminatedSlaves);
+    }
+    if ('IdleShutdown' in convertedConfigs) {
+      convertedConfigs.IdleShutdown = Number.parseInt(convertedConfigs.IdleShutdown, 10);
+    }
+    if ('ResourceTracker' in convertedConfigs) {
+      convertedConfigs.ResourceTracker = this.convertToBoolean(convertedConfigs.ResourceTracker);
+    }
+    if ('StaggerInstances' in convertedConfigs) {
+      convertedConfigs.StaggerInstances = this.convertToBoolean(convertedConfigs.StaggerInstances);
+    }
+    if ('StrictHardCap' in convertedConfigs) {
+      convertedConfigs.StrictHardCap = this.convertToBoolean(convertedConfigs.StrictHardCap);
+    }
+    if ('UseLocalCredentials' in convertedConfigs) {
+      convertedConfigs.UseLocalCredentials = this.convertToBoolean(convertedConfigs.UseLocalCredentials);
+    }
+
+    let configs = [];
+    for (const [key, value] of Object.entries(pluginOptions)) {
+      if (value !== undefined) {
+        configs.push({
+          Key: key,
+          Value: value,
+        });
+      }
+    }
+    return configs;
+  }
+
+  /**
+   * Retrieve CA certificate data from the Secret with the given ARN.
+   * @param certificateArn
+   */
+  protected async readCertificateData(certificateArn: string): Promise<string> {
+    const data = await Secret.fromArn(certificateArn, this.secretsManagerClient).getValue();
+    if (Buffer.isBuffer(data) || !/BEGIN CERTIFICATE/.test(data as string)) {
+      throw new Error(`CA Certificate Secret (${certificateArn}) must contain a Certificate in PEM format.`);
+    }
+    return data as string;
```

Comment:
I took this from our mongo db post install implementation. Instead of just copy/pasting it to my class, I now moved this function into a separate module, so it can be used in both places without duplication.

But as the mongo db relies on the current implementation of this function, I wouldn't like to change it. 
I could add there a second function with `Buffer` support if you think this is really necessary.

Reply:
We can leave this out of scope for this PR. I believe this would be a very rare edge-case where a customer does not correctly wire up the cert secrets to the RQ. The impact is ungraceful error handling

Resolving.